optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/valuerange/buildin/temporal/TemporalValueRangeTest.java

```diff
@@ -332,4 +359,118 @@ public void getAndCreateOriginalIteratorInSyncForLocalDate() {
+        assertTrue(range.contains(from));
+        assertFalse(range.contains(to));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullFrom() {
+        assertNotNull(new TemporalValueRange<>(null, LocalDate.MAX, 1, ChronoUnit.DAYS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullTo() {
+        assertNotNull(new TemporalValueRange<>(LocalDate.MIN, null, 1, ChronoUnit.DAYS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void nullIncrementUnitType() {
+        assertNotNull(new TemporalValueRange<>(LocalDate.MIN, LocalDate.MAX, 1, null));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void invalidIncrementUnitAmount() {
+        assertNotNull(new TemporalValueRange<>(LocalDate.MIN, LocalDate.MAX, 0, ChronoUnit.DAYS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void fromAfterTo() {
+        assertNotNull(new TemporalValueRange<>(LocalDate.MAX, LocalDate.MIN, 1, ChronoUnit.DAYS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void incrementUnitTypeNotSupportedByFrom() {
+        TemporalMock from = mock(TemporalMock.class);
+        when(from.isSupported(any(TemporalUnit.class))).thenReturn(Boolean.FALSE);
+        TemporalMock to = mock(TemporalMock.class);
+        when(from.isSupported(any(TemporalUnit.class))).thenReturn(Boolean.TRUE);
+        assertNotNull(new TemporalValueRange<>(from, to, 1, mock(TemporalUnit.class)));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void incrementUnitTypeNotSupportedByTo() {
+        TemporalMock from = mock(TemporalMock.class);
+        when(from.isSupported(any(TemporalUnit.class))).thenReturn(Boolean.TRUE);
+        TemporalMock to = mock(TemporalMock.class);
+        when(from.isSupported(any(TemporalUnit.class))).thenReturn(Boolean.FALSE);
+        assertNotNull(new TemporalValueRange<>(from, to, 1, mock(TemporalUnit.class)));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void remainderOnIncrementAmount() {
+        assertNotNull(new TemporalValueRange<>(Year.of(0), Year.of(3), 2, ChronoUnit.YEARS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void remainderOnIncrementType() {
+        LocalTime from = LocalTime.of(11, 30);
+        LocalTime to = LocalTime.of(13, 29);
+        assertNotNull(new TemporalValueRange<>(from, to, 1, ChronoUnit.HOURS));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void remainderOnIncrementTypeExceedsMaximumYear() {
+        Year from = Year.of(Year.MIN_VALUE);
+        Year to = Year.of(Year.MAX_VALUE - 0);
+        assertNotEquals(0, (to.getValue() - from.getValue()) % 10); // Maximum Year range is not divisible by 10
+        assertNotNull(new TemporalValueRange<>(from, to, 1, ChronoUnit.DECADES));
+    }
```

Comment:
This test would also throw `DateTimeException` without the fix. Now it "succeeds" with IllegalArgException explaining that Decades increment doesn't fit into the `[from, to)` by integer number of times.